clients/src/test/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessorTest.java

```diff
@@ -118,13 +118,12 @@ private static Stream<Arguments> applicationEvents() {
         final long currentTimeMs = 12345;
         return Stream.of(
                 Arguments.of(new PollEvent(100)),
-                Arguments.of(new NewTopicsMetadataUpdateRequestEvent()),
                 Arguments.of(new AsyncCommitEvent(new HashMap<>())),
                 Arguments.of(new SyncCommitEvent(new HashMap<>(), 500)),
                 Arguments.of(new ResetPositionsEvent(500)),
                 Arguments.of(new ValidatePositionsEvent(500)),
                 Arguments.of(new TopicMetadataEvent("topic", Long.MAX_VALUE)),
-                Arguments.of(new AssignmentChangeEvent(offset, currentTimeMs)));
+                Arguments.of(new AssignmentChangeEvent(12345, 12345, Collections.emptyList())));
```

Comment:
note that by adding this we're only testing that the call to `process(AppEvent)` ends up triggering the `process(AssignmentChangeEvent)`, nothing more. So should we add a test for the `process(AssignmentChangeEvent)`? We should ensure that it takes the actions we expect (commit, assign, request metadata update), using a `processor` instance, not a mock.

Reply:
I add `testAssignmentChangeEvent` and `testAssignmentChangeEventWithException` to check `processor` instance. Thank you.